base/loading.jl

```diff
@@ -3047,6 +3047,14 @@ function module_build_id(m::Module)
     return (UInt128(hi) << 64) | lo
 end
 
+function object_build_id(obj)
+    mod = ccall(:jl_object_top_module, Any, (Any,), obj)
+    if mod === nothing
+        return nothing
+    end
+    return module_build_id(mod::Module)
```

Comment:
Maybe we should only use the lower 64bit since the upper is the cache sum that isn't available during precompilation.

```
julia> Base.module_build_id(TPkg)
0xfafbfcfd3b096e5c000002019366584f

julia> TPkg.bid
0xffffffffffffffff000002019366584f
```